app/controllers/account_details.js

```diff
@@ -0,0 +1,111 @@
+import Ember from "ember";
+import AjaxPromise from 'browse/utils/ajax-promise';
+const { getOwner } = Ember;
+
+export default Ember.Controller.extend({
+  queryParams: ['orgId'],
+
+  authenticate:Ember.inject.controller(),
+  messageBox: Ember.inject.service(),
+  i18n: Ember.inject.service(),
+  organisationId: Ember.computed.alias('model.organisation.id'),
+  organisationsUserId: Ember.computed.alias('model.organisationsUser.id'),
+  position: "",
+
+  userTitle: Ember.computed('model', function() {
+    let userTitle = this.get('model.user.title');
+    let titles = this.get('titles');
+
+    if(userTitle) {
+      let filteredUserTitle = titles.filter((item) => userTitle === item.id);
+      return { name: filteredUserTitle[0].name.string, id: userTitle };
+    } else {
+      return { name: titles.get('firstObject.name').string, id: 'Mr' };
+    }
+  }),
+
+  selectedTitle: Ember.computed('userTitle', function (){
+    return{ name: this.get('userTitle.name'), id: this.get('userTitle.id')};
+  }),
+
+  titles: Ember.computed(function() {
+    let mr = this.get("i18n").t("account.user_title.mr");
+    let mrs = this.get("i18n").t("account.user_title.mrs");
+    let miss = this.get("i18n").t("account.user_title.miss");
+    let ms = this.get("i18n").t("account.user_title.ms");
+
+    return [
+      { name: mr, id: "Mr" },
+      { name: mrs, id: "Mrs" },
+      { name: miss, id: "Miss" },
+      { name: ms, id: "Ms" }
+    ];
+  }),
+
+  redirectToTransitionOrBrowse() {
+    var attemptedTransition = this.get('authenticate').get('attemptedTransition');
+    if (attemptedTransition) {
+      this.set('attemptedTransition', null);
+      attemptedTransition.retry();
+    } else {
+      this.transitionToRoute("browse");
+    }
+  },
+
+  organisationsUserParams() {
+    var organisationsUserId = this.get('organisationsUserId');
+    var user = this.get('model.user');
+    var position = organisationsUserId ? this.get('model.organisationsUser.position') : this.get('position');
+    var title = this.get('selectedTitle.id');
+    var params = {
+      organisation_id: this.get('organisationId'),
+      position: position,
+      user_attributes: {
+        first_name: user.get('firstName'),
+        last_name: user.get('lastName'),
+        mobile: user.get('mobile'),
+        email: user.get('email'),
+        title: title
+      }
+    };
+    if (organisationsUserId) {
+      params.id = organisationsUserId;
+    }
+
+    return params;
+  },
+
+  saveOrUpdateAccount(url, actionType) {
```

Comment:
This should be an action instead of function.
Function should be only written to help actions execute.
Functions shouldn't fire requests to API, that's what actions are for.